aea/skills/base.py

```diff
@@ -559,6 +559,7 @@ def __init__(
         self.handlers = handlers if handlers is not None else {}
         self.behaviours = behaviours if behaviours is not None else {}
         self.models = models if models is not None else {}
+        self.skill_context._skill = self
```

Comment:
@marcofavorito this seems more consistent as it attaches the skill to the context no matter where the skill is initialized

Reply:
ok